pandas/core/arrays/base.py

```diff
@@ -603,6 +603,16 @@ def isna(self) -> np.ndarray | ExtensionArraySupportsAnyAll:
         """
         raise AbstractMethodError(self)
 
+    @property
```

Comment:
can we not use a cached_property?

Reply:
no, EAs are mutable

Reply:
right (though actually we *could* do this as we know when they *are* mutated, e.g keep a dirty bit). but out of scope

Reply:
i've looked into this w/r/t `DatetimeArray.freq` and found that propagating it through views (e.g. `arr[:5][0] = np.nan`) makes this a giant pain